.github/workflows/windows.yml

```diff
@@ -46,42 +46,154 @@ env:
   RUSTUP_WINDOWS_PATH_ADD_BIN: 1
 
 jobs:
+  # Automatic runner selection for job: build
+  # Runs the underlying job (“workload”) on a self-hosted runner if available,
+  # with the help of a `runner-select` job and a `runner-timeout` job.
+
+  # Selects a self-hosted runner if available, or else a GitHub-hosted runner.
+  # We generate a unique id for the workload, find an idle self-hosted runner
+  # with the given `image:` label that wasn’t already reserved by another
+  # `runner-select` job run, and reserve it with a `reserved-for:<id>` label.
+  runner-select:
+    name: Runner select
+    runs-on: ubuntu-latest
+    outputs:
+      unique-id: ${{ steps.select.outputs.unique_id }}
+      selected-runner-label: ${{ steps.select.outputs.selected_runner_label }}
+      is-self-hosted: ${{ steps.select.outputs.is_self_hosted }}
+    concurrency:
+      group: servo-reserve-self-hosted-runner
+      cancel-in-progress: false
+    permissions: write-all
+    steps:
+      - name: Select and reserve best available runner
+        id: select
+        # Set the variables below to your desired runner images, runner scope
+        # (org or repo), and a token with permission to write to that scope.
+        run: |
+          github_hosted_runner_label=windows-2022
+          self_hosted_image_label=self-hosted-image:windows10
+          self_hosted_runner_scope=/orgs/${{ github.repository_owner }}/actions/runners
+          export GH_TOKEN=${{ secrets.RUNNER_API_TOKEN }}
```

Comment:
This causes `GH` to fail if secrets is not provided (in personal forks). In that case we want to run on github runner.

See https://github.com/sagudev/servo/actions/runs/10506866977/job/29107421168

Reply:
great catch! fixed in a2535502d4f3ef2b11225d785a97fb39fa62f2dd